src/traces/scatter3d/convert.js

```diff
@@ -133,14 +133,49 @@ function calculateErrorParams(errors) {
     return {capSize: capSize, color: color, lineWidth: lineWidth};
 }
 
+function parseAlignmentX(a) {
+    if(a === null || a === undefined) return 0;
+    else if(typeof(a) === 'number') return a;
```

Comment:
Why would `textposition[i]` ever be a number here?

Reply:
Well, I thought we could somehow make use of other numbers (instead than only -1|0|1) to scale the distances. But yeah I may remove those condition checks (also for null & undefined) since I noticed the inputs seem to be parsed and validated before getting to there?